compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8089,6 +8089,7 @@ public void visit(BLangQueryExpr queryExpr) {
         BLangStatementExpression stmtExpr = queryDesugar.desugar(queryExpr, env, getVisibleXMLNSStmts(env));
         result = rewrite(stmtExpr, env);
         this.isVisitingQuery = prevIsVisitingQuery;
+        queryExpr.desugared = true;
```

Comment:
What was the reason this is not set through the normal flow?

Reply:
`result` is `statement expression` not the `query expression`. So, the `desugared` flag is to `statement expression` not to the `query expression`.

Reply:
If that's the case shouldn't the `desugared=true` be set to the input node not the result node in places like https://github.com/ballerina-platform/ballerina-lang/blob/bac71a8061bd6cf6f925e6ba2f27361443057969/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L8376?